api/trading_data_v2.go

```diff
@@ -193,6 +195,143 @@ func entityMarketDataListToProtoList(list []entities.MarketData) (*v2.MarketData
 	return &connection, nil
 }
 
+func (t *tradingDataServiceV2) ObserveMarketsDepth(req *v2.ObserveMarketsDepthRequest, srv v2.TradingDataService_ObserveMarketsDepthServer) error {
```

Comment:
not blocking: seems like we are making this api able to observer multiple market depth at the same time? Is that necessary? Does it not just complexify things a little either for us or even the client?

Reply:
there is no real additional complexity on the data node side to support this, but it does mean that if a client wants to subscribe to 20 out of 40 markets say, that will be just one grpc stream as opposed to twenty streams.  With the current api the solution is one grpc stream per market, or you subscribe with empty string and get updates for every market in the system in one stream.

Reply:
right, that's my point. is subscribing to all markets, or many of them a behaviour we want to encourage? I imagine it can put some stress on the system, and we may want to avoid that? I can be totally wrong, and feel free to close this conversation.

Reply:
For level 1 data the current choice is get data for all markets over one stream and then filter on the client, or open a stream  per market, so it definetely makes sense in this case as it would be quite common to need this on an instrument watch or similar.    Its possible that for depth, certainly for a GUI, more than one depth subscription at a time is not really needed.  If we think its reasonable that we will have clients that will want to subscribe to the depth of more than one market concurrently, then I would say this is the better API, it will use less resource than having to open a subscription per market.  In practice how likely clients will want to do this, I don't know.